### DISCUSSION TITLE
Stabilization problem in simulation.

### DISCUSSION THREAD

**Discussion by SupSonAlv** [GuidodiPasquo/Actively-Stabilized-Model-Rocket-Simulator#13] [General]
Hi  
at AeroVECTOR
We wanted to do some small tests. We tried to use the python example in the examples, why does it behave as if it has no effect on the stability of the rocket, even though we change the PID values and wing sizes there.
What is the solution to this problem?
![image](https://github.com/GuidodiPasquo/AeroVECTOR/assets/138517098/bdd1bf00-886a-4cb7-a919-e23dfa12374d)
![image](https://github.com/GuidodiPasquo/AeroVECTOR/assets/138517098/9a7e0090-5339-4e1c-a764-0f6c1edd00ee)

**Comment by GuidodiPasquo** [comment]
Hello!
Indeed, the PID values do nothing there, you have to go to AeroVECTOR-master\3 - Examples\Example Python SITL\SITL Modules and edit example_python_sitl.py to change the PID parameters. I just tested the fins and it changes the behavior of the rocket, so I couldn't reproduce the error.

I did got weird behaviors with the released version, please download the source code and test if you still have those problems

**Reply by SupSonAlv** [reply]
Does the data changed and saved in the 'Sim setup' tab have no effect on the simulation?
- I downloaded the shared code again. But even if I reduce the wing dimensions, I can't see the effect in the simulation.
-Is it not possible to set values directly from here?

When you change the data here, the effect on the simulation is seen.
![image](https://github.com/GuidodiPasquo/AeroVECTOR/assets/138517098/447c03ff-261d-4ec8-a14c-2f67719a9fc2)


Finally, after the simulation is over, is there no other way to make a simulation with new parameters again, other than closing and reopening the application?

**Reply by GuidodiPasquo** [reply]
The PID controller in 'Sim Setup', when using SITL, is disabled. The initial parameters and so on are taken into account in the simulation.

I couldn't reproduce the problem with the fins. When I put the new dimensions and update the fins, they show up correctly in the simulation.

Exactly, when you change the parameters there, they affect the simulation because the program is loading the controller in that Python file. 

After you click 'Finish', you should be able to open example_python_sitl.py, change and save the parameters, and run the simulation again without closing the application.

**Reply by SupSonAlv** [reply]
The problem is now solved. The situation is as you said.
If a new model is created directly on AeroVECTOR without starting the STL, the values entered in the program's interface are taken as basis.
